### DISCUSSION TITLE
2-way communication

### DISCUSSION THREAD

**Discussion by Jens-Wymeersch** [tsightler/ring-mqtt#750] [Feature Requests]
Most, if not all cameras have the ability, to communicate back. Obviously, with the RTSP feed this isn't possible. Would be great to have this feature in like the Homebridge app (https://github.com/dgreif/ring)

**Comment by tsightler** [comment]
This has nothing to do with RTSP, it can support two-way audio just fine via ONVIF-T, actually the vast majority of devices that support two-way audio do so via RTSP with ONVIF-T.  However, unlike HomeKit, Home Assistant has no native support for two-way audio.  There are a few third-party ways to get two-way audio support in HA via custom components like the Frigate or WebRTC cards in concert with go2rtc, but these don't work with ring-mqtt today because I haven't implemented ONVIF-T return channel audio.  This is due mostly due to lack of time but also because there is limited use case as, realistically, these methods offer llittle difference from just embedding the Ring Web dashboard into HA since, at that point, everything is basically outside of HA anyway and it's just some custom cards which will do it.

I'm happy to accept PRs that implement support for two-way audio in ring-mqtt but it's unlikely I will be personally adding it any time soon.

**Reply by Jens-Wymeersch** [reply]
You have done an awesome job and effectively time is always an issue...
My use case is the following, I have smart locks (Nuki) and smart gates (iSmartgate) which obviously don't natively integrate with Ring. So to have an intercom experience, it would have been great to combine the 2 like homekit does. But without 2-way audio communication, this is a bit tough.

**Comment by Jens-Wymeersch** [comment]
@tsightler I've just tried to embedded to the ring web dashboard. Ring doesn't allow me to do this (iframe). Any other ideas or change of mind ?

**Reply by tsightler** [reply]
There is nothing to change my mind about, as I said above, happy to review a PR that implements 2-way audio, but the chances of me ever coding it are near zero any time soon as it is a lot of work for very little value and nothing but trouble from users who won't understand all of the limitations.
